kotsadm/web/src/components/apps/AppSnapshotRestore.jsx

```diff
@@ -5,27 +5,81 @@ import Helmet from "react-helmet";
 import { Line } from "rc-progress";
 import Loader from "../shared/Loader";
 import { Utilities } from "@src/utilities/utilities";
-import { restoreDetail } from "../../queries/SnapshotQueries";
+import { Repeater } from "@src/utilities/repeater";
 import { cancelRestore } from "../../mutations/SnapshotMutations";
 import "../../scss/components/snapshots/AppSnapshots.scss";
 
 class AppSnapshotRestore extends Component {
   state = {
+    fetchRestoreDetailJob: new Repeater(),
+    loadingRestoreDetail: true,
+    restoreDetail: {},
+    errorMessage: "",
+    errorTitle: "",
+
     cancelingRestore: false,
     cancelRestoreErr: "",
-    cancelRestoreErrorMsg: ""
+    cancelRestoreErrorMsg: "",
   }
 
   componentDidMount() {
-    this.props.restoreDetail.startPolling(2000);
+    this.state.fetchRestoreDetailJob.start(this.fetchRestoreDetail, 2000);
   }
 
   componentDidUpdate(lastProps) {
-    if (this.props.restoreDetail?.restoreDetail !== lastProps.restoreDetail?.restoreDetail && this.props.restoreDetail?.restoreDetail) {
-      const phase = this.props.restoreDetail?.restoreDetail?.phase;
-      if (phase !== "New" && phase !== "InProgress") {
-        this.props.restoreDetail.stopPolling();
+    const { match } = this.props;
+    if (match.params.id !== lastProps.match.params.id) {
+      this.state.fetchRestoreDetailJob.stop();
+      this.state.fetchRestoreDetailJob.start(this.fetchRestoreDetail, 2000);
+    } else {
+      const phase = this.state.restoreDetail?.phase;
+      if (phase && phase !== "New" && phase !== "InProgress") {
+        this.state.fetchRestoreDetailJob.stop();
+      }
+    }
+  }
+
+  fetchRestoreDetail = async () => {
+    const { match } = this.props;
+    const restoreName = match.params.id;
+
+    this.setState({
+      errorMessage: "",
+      errorTitle: "",
+    });
+
+    try {
+      const res = await fetch(`${window.env.API_ENDPOINT}/app/${this.props.app?.slug}/snapshot/restore/${restoreName}`, {
+        method: "GET",
+        headers: {
+          "Authorization": Utilities.getToken(),
+        }
+      });
+      if (!res.ok) {
```

Comment:
should log out on 401 status code

Reply:
can you send me an example?

Reply:
https://github.com/replicatedhq/kots/blob/5a6baa27c54e10529b88f73c860f127f64ff8ee8/kotsadm/web/src/Root.jsx#L151